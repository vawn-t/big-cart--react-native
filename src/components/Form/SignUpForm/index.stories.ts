import type { Meta, StoryObj } from '@storybook/react-native';

import SignUp from '.';

const meta: Meta<typeof SignUp> = {
  title: 'Components/SignUp',
  component: SignUp,
};

export default meta;

type Story = StoryObj<React.ComponentProps<typeof SignUp>>;

export const Base: Story = {
  args: {},
};
