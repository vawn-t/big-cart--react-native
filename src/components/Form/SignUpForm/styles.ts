import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    rowGap: 5,
  },
  submitButton: {
    marginTop: 10,
  },
});

export default styles;
