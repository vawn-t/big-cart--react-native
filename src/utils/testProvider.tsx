import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { RenderOptions, render } from '@testing-library/react-native';

const Stack = createNativeStackNavigator();
const queryClient = new QueryClient();

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const component = () => <>{children}</>;

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="test">{component}</Stack.Screen>
        </Stack.Navigator>
      </NavigationContainer>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  // eslint-disable-next-line react/display-name
  return ({ children }: { children: React.ReactNode }) => {
    const component = () => <>{children}</>;
    return (
      <QueryClientProvider client={queryClient}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="test">{component}</Stack.Screen>
          </Stack.Navigator>
        </NavigationContainer>
      </QueryClientProvider>
    );
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render, createWrapper };
