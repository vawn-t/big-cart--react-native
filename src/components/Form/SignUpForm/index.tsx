import { useCallback, useRef, useState } from 'react';
import { TextInput, View } from 'react-native';

// Components
import { Button, Input } from '@components/common';

// Types
import { IconName, SignUpFormData, SignUpFormErrors } from '@interfaces/index';

// Utils
import { isDisableSubmitButton } from '@utils/index';

// Hooks
import useAuthenticator from '@hooks/useAuthenticator';

// Styles
import styles from './styles';

const SignUpForm = () => {
  const [formData, setFormData] = useState<SignUpFormData>({
    email: '',
    phone: '',
    password: '',
  });

  // Refs
  const emailRef = useRef<TextInput>(null);
  const phoneRef = useRef<TextInput>(null);
  const passwordRef = useRef<TextInput>(null);

  // Hooks
  const {
    isLoading,
    errors,
    clearEmailError,
    clearPasswordError,
    clearPhoneNumberError,
    signUp,
  } = useAuthenticator();

  const handleSubmitForm = () => {
    emailRef.current?.blur();
    passwordRef.current?.blur();

    signUp(formData);
  };

  const handleFocusEmail = useCallback(() => {
    clearEmailError();
  }, []);

  const handleFocusPassword = useCallback(() => {
    clearPasswordError();
  }, []);

  const handleFocusPhone = useCallback(() => {
    clearPhoneNumberError();
  }, []);

  return (
    <View style={styles.container}>
      <Input
        ref={emailRef}
        icon={IconName.Email}
        placeholder="Email address"
        keyboardType="email-address"
        isError={!!errors.email}
        onChangeText={(email) => setFormData((prev) => ({ ...prev, email }))}
        onFocus={handleFocusEmail}
      />

      <Input
        ref={phoneRef}
        icon={IconName.Phone}
        placeholder="Phone number"
        keyboardType="numeric"
        isError={!!(errors as SignUpFormErrors).phoneNumber}
        onChangeText={(phone) => setFormData((prev) => ({ ...prev, phone }))}
        onFocus={handleFocusPhone}
      />

      <Input
        ref={passwordRef}
        icon={IconName.Password}
        placeholder="● ● ● ● ● ●"
        autoComplete="password"
        isError={!!errors.password}
        onChangeText={(password) =>
          setFormData((prev) => ({ ...prev, password }))
        }
        onFocus={handleFocusPassword}
      />

      <Button
        title="Signup"
        style={styles.submitButton}
        onPress={handleSubmitForm}
        disabled={isDisableSubmitButton(formData) || isLoading}
      />
    </View>
  );
};

export default SignUpForm;
