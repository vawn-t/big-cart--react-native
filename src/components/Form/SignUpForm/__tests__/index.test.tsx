import SignUpForm from '..';
import { fireEvent, render, screen, waitFor } from '@utils/testProvider';
import { colors } from '@themes/index';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

jest.mock('@hooks/useAuth/useAuthSignUp', () => {
  return jest.fn().mockReturnValue({
    isSuccess: true,
    data: { token: 'token' },
    variables: { email: 'test@gmail.com' },
  });
});

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

const setup = () => {
  return render(<SignUpForm />);
};

describe('SignUpForm', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('renders the form correctly', () => {
    const { getByPlaceholderText, toJSON } = setup();

    expect(toJSON()).toMatchSnapshot();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(getByPlaceholderText('Phone number')).toBeTruthy();
    expect(getByPlaceholderText('● ● ● ● ● ●')).toBeTruthy();
  });

  it('should update error style if submit invalid data', async () => {
    const { getByText } = setup();

    fireEvent.changeText(screen.getByPlaceholderText('Email address'), 'email');
    fireEvent.changeText(
      screen.getByPlaceholderText('Phone number'),
      '0123456789',
    );
    fireEvent.changeText(
      screen.getByPlaceholderText('● ● ● ● ● ●'),
      'password',
    );

    fireEvent.press(getByText('Signup'));

    await waitFor(() => {
      const errorInputs = screen.getAllByTestId('Input');
      expect(errorInputs[0]).toHaveStyle({
        borderColor: colors.red,
      });
      expect(errorInputs[1]).toHaveStyle({
        borderColor: colors.red,
      });
      expect(errorInputs[2]).toHaveStyle({
        borderColor: colors.red,
      });
    });
  });
});
