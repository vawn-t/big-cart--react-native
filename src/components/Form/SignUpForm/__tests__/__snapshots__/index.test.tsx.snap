// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders the form correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "rowGap": 5,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                  null,
                ]
              }
              testID="Input"
            >
              <View
                style={
                  {
                    "flex": 1,
                    "paddingHorizontal": 28,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="19"
                  bbWidth="23"
                  fill="none"
                  focusable={false}
                  height={19}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 19,
                        "width": 23,
                      },
                    ]
                  }
                  width={23}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M20.979.523H2.02A2.025 2.025 0 0 0 0 2.545V16.02c0 1.111.904 2.022 2.021 2.022h18.957c1.111 0 2.022-.904 2.022-2.021V2.544c0-1.11-.904-2.022-2.021-2.022Zm-.28 1.348-9.156 9.157L2.307 1.87H20.7ZM1.348 15.742V2.818l6.49 6.434-6.49 6.49Zm.953.953 6.494-6.494 2.275 2.256c.264.261.69.26.951-.002l2.22-2.219 6.458 6.46H2.301Zm19.351-.953-6.459-6.459 6.46-6.459v12.918Z"
                      fill={
                        {
                          "payload": 4287006857,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <TextInput
                autoComplete="off"
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Email address"
                secureTextEntry={false}
                style={
                  [
                    {
                      "flex": 10,
                      "paddingVertical": 20,
                    },
                    undefined,
                  ]
                }
                testID="Input-TextInput"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flex": 1,
                    "opacity": 1,
                    "paddingHorizontal": 28,
                  }
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                  null,
                ]
              }
              testID="Input"
            >
              <View
                style={
                  {
                    "flex": 1,
                    "paddingHorizontal": 28,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="23"
                  bbWidth="23"
                  fill="none"
                  focusable={false}
                  height={23}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 23,
                        "width": 23,
                      },
                    ]
                  }
                  width={23}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup
                      clipPath="a"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGPath
                        d="M4.643 15.289c2.269 2.712 5 4.847 8.116 6.357 1.187.563 2.774 1.23 4.542 1.345.11.004.215.009.324.009 1.187 0 2.14-.41 2.917-1.253a.104.104 0 0 0 .02-.024c.276-.334.59-.634.919-.953.224-.215.453-.439.672-.668 1.015-1.058 1.015-2.402-.01-3.426L19.28 13.81c-.486-.505-1.067-.772-1.677-.772-.61 0-1.197.267-1.697.768l-1.706 1.706c-.158-.09-.32-.172-.472-.248a5.88 5.88 0 0 1-.524-.286c-1.554-.986-2.965-2.273-4.313-3.927-.682-.863-1.14-1.587-1.459-2.326.448-.405.868-.829 1.273-1.244.143-.147.29-.295.438-.443.515-.515.791-1.11.791-1.716 0-.605-.271-1.2-.79-1.715l-1.421-1.42c-.167-.167-.324-.33-.486-.496a26.133 26.133 0 0 0-.968-.958C5.778.253 5.201 0 4.591 0c-.605 0-1.187.253-1.697.739L1.112 2.52A3.662 3.662 0 0 0 .02 4.866c-.09 1.139.12 2.35.663 3.813.834 2.263 2.092 4.365 3.96 6.61ZM1.183 4.966a2.52 2.52 0 0 1 .758-1.62l1.773-1.773c.276-.267.581-.405.877-.405.29 0 .586.138.858.414.319.296.62.606.943.934.162.167.33.334.496.506l1.42 1.42c.296.295.448.596.448.891 0 .295-.152.596-.448.891-.148.148-.295.3-.443.448-.443.448-.858.872-1.315 1.277l-.024.024c-.396.396-.334.772-.239 1.058l.015.038c.367.882.877 1.72 1.673 2.722 1.43 1.763 2.935 3.13 4.594 4.184.205.134.424.238.629.343.19.096.367.186.524.286.02.01.034.02.053.029.157.08.31.119.462.119.381 0 .629-.243.71-.324l1.783-1.783c.276-.276.576-.424.872-.424.362 0 .657.224.843.425l2.874 2.869c.572.571.567 1.191-.014 1.796-.2.215-.41.42-.634.634-.334.324-.682.658-.996 1.034-.548.591-1.201.868-2.045.868-.08 0-.167-.005-.248-.01-1.563-.1-3.016-.71-4.108-1.23a23.18 23.18 0 0 1-7.725-6.052c-1.778-2.14-2.974-4.132-3.765-6.267-.491-1.31-.677-2.364-.6-3.322Z"
                        fill={
                          {
                            "payload": 4287006857,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                    </RNSVGGroup>
                    <RNSVGDefs>
                      <RNSVGClipPath
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        name="a"
                      >
                        <RNSVGPath
                          d="M0 0h23v23H0z"
                          fill={
                            {
                              "payload": 4294967295,
                              "type": 0,
                            }
                          }
                          propList={
                            [
                              "fill",
                            ]
                          }
                        />
                      </RNSVGClipPath>
                    </RNSVGDefs>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <TextInput
                autoComplete="off"
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Phone number"
                secureTextEntry={false}
                style={
                  [
                    {
                      "flex": 10,
                      "paddingVertical": 20,
                    },
                    undefined,
                  ]
                }
                testID="Input-TextInput"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flex": 1,
                    "opacity": 1,
                    "paddingHorizontal": 28,
                  }
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                  null,
                ]
              }
              testID="Input"
            >
              <View
                style={
                  {
                    "flex": 1,
                    "paddingHorizontal": 28,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="23"
                  bbWidth="18"
                  fill="none"
                  focusable={false}
                  height={23}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 23,
                        "width": 18,
                      },
                    ]
                  }
                  width={18}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M15.094 23H2.156A2.159 2.159 0 0 1 0 20.844V10.78c0-1.19.968-2.156 2.156-2.156h12.938c1.188 0 2.156.967 2.156 2.156v10.063c0 1.19-.968 2.156-2.156 2.156ZM2.156 10.062a.72.72 0 0 0-.719.72v10.062a.72.72 0 0 0 .72.718h12.937a.72.72 0 0 0 .719-.718V10.78a.72.72 0 0 0-.72-.719H2.157Z"
                      fill={
                        {
                          "payload": 4287006857,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M13.656 10.063a.719.719 0 0 1-.719-.72V5.75a4.318 4.318 0 0 0-4.312-4.313A4.318 4.318 0 0 0 4.312 5.75v3.594a.719.719 0 0 1-1.437 0V5.75A5.756 5.756 0 0 1 8.625 0a5.756 5.756 0 0 1 5.75 5.75v3.594a.719.719 0 0 1-.719.719Z"
                      fill={
                        {
                          "payload": 4287006857,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <TextInput
                autoComplete="password"
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="● ● ● ● ● ●"
                secureTextEntry={true}
                style={
                  [
                    {
                      "flex": 10,
                      "paddingVertical": 20,
                    },
                    undefined,
                  ]
                }
                testID="Input-TextInput"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "flex": 1,
                    "opacity": 1,
                    "paddingHorizontal": 28,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="17"
                  bbWidth="27"
                  fill="none"
                  focusable={false}
                  height={17}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 17,
                        "width": 27,
                      },
                    ]
                  }
                  width={27}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M26.314 7.924C26.078 7.601 20.441 0 13.241 0S.404 7.6.168 7.924a.874.874 0 0 0 0 1.03c.236.324 5.874 7.925 13.073 7.925 7.2 0 12.837-7.6 13.073-7.924a.873.873 0 0 0 0-1.03Zm-13.073 7.209c-5.303 0-9.896-5.045-11.256-6.694 1.358-1.65 5.942-6.693 11.256-6.693 5.303 0 9.896 5.044 11.256 6.694-1.358 1.65-5.941 6.693-11.256 6.693Z"
                      fill={
                        {
                          "payload": 4287006857,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M13.241 3.879a4.566 4.566 0 0 0-4.56 4.56A4.566 4.566 0 0 0 13.24 13a4.566 4.566 0 0 0 4.56-4.56 4.566 4.566 0 0 0-4.56-4.56Zm0 7.6a3.044 3.044 0 0 1-3.04-3.04 3.044 3.044 0 0 1 3.04-3.04 3.044 3.044 0 0 1 3.04 3.04 3.044 3.044 0 0 1-3.04 3.04Z"
                      fill={
                        {
                          "payload": 4287006857,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "disabled": true,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#6CC51D",
                    "borderRadius": 5,
                    "flexDirection": "row",
                  },
                  {
                    "opacity": 0.5,
                  },
                  {
                    "marginTop": 10,
                  },
                ]
              }
            >
              <View
                style={
                  {
                    "flexGrow": 1,
                    "paddingVertical": 20,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                      },
                      {
                        "fontFamily": "Poppins-SemiBold",
                      },
                      {
                        "fontSize": 15,
                      },
                      {
                        "color": "#FFFFFF",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Signup
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
